src/main/java/leets/leets_mate/domain/GroupGenerator.java

```diff
@@ -0,0 +1,15 @@
+package leets.leets_mate.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GroupGenerator {
+    public List<List<String>> generateRandomGroups(Members members, MaxGroupSize maxGroupSize) {
+        members.shuffleMembers();
+        List<List<String>> matchedGroups = new ArrayList<>();
+        for (int i = 0; i < members.getCount(); i += maxGroupSize.maxGroupSize) {
+            matchedGroups.add(members.membersList.subList(i, Math.min(i + maxGroupSize.maxGroupSize, members.getCount())));
```

Comment:
`List<List<String>>`가 무엇인지 이해하기 어렵네요. 클래스로 만들어 이름을 붙여주는 건 어떨까요?
혹은 
```suggestion
            List<String> matchedGroup = members.membersList.subList(i, Math.min(i + maxGroupSize.maxGroupSize, members.getCount()));
            matchedGroups.add(matchedGroup);
```
중요한 부분은 지역변수로 선언해 강조하는 방법도 있겠네요!